apis/v1alpha2/httproute_types.go

```diff
@@ -408,6 +408,10 @@ type HTTPQueryParamMatch struct {
 	Value string `json:"value"`
 }
 
+// HTTPMethod describes how to select a HTTP route by matching the HTTP
+// method.
+type HTTPMethod string
```

Comment:
Is it an appropriate layer to add some kind of enum to allow specific values? E.g. the http methods defined by the HTTP spec.

See https://datatracker.ietf.org/doc/html/rfc7231#section-4 and here https://datatracker.ietf.org/doc/html/rfc5789#section-2

I thinking more about a way to avoid mistakes when defining the paths and only when the config is applied to not route through to the correct service

Reply:
Alternatively we could reuse the HttpMethod from the `genproto` package https://pkg.go.dev/google.golang.org/genproto/googleapis/cloud/scheduler/v1#HttpMethod

Reply:
I like the idea of an enum here. I'd also recommend adding enum validation to the field as shown here: https://github.com/kubernetes-sigs/gateway-api/blob/master/apis/v1alpha2/httproute_types.go#L271. 

I think I'd prefer defining our own custom types instead of reusing the ones defined in genproto. That will allow us more flexibility in terms of documentation and support levels.

Also appreciate the RFC references by @brunoarueira. Where possible, we like to reference specific RFCs when documenting a feature/field in the comments (example: https://github.com/kubernetes-sigs/gateway-api/blob/master/apis/v1alpha2/httproute_types.go#L342).

Reply:
FWIW there is an IANA registry for HTTP methods

https://www.iana.org/assignments/http-methods/http-methods.xhtml